src/main/java/com/commercetools/sync/shoppinglists/utils/ShoppingListSyncUtils.java

```diff
@@ -0,0 +1,85 @@
+import static com.commercetools.sync.shoppinglists.utils.ShoppingListUpdateActionUtils.buildSetDeleteDaysAfterLastModificationUpdateAction;
+import static com.commercetools.sync.shoppinglists.utils.ShoppingListUpdateActionUtils.buildSetDescriptionUpdateAction;
+import static com.commercetools.sync.shoppinglists.utils.ShoppingListUpdateActionUtils.buildSetSlugUpdateAction;
+
+public final class ShoppingListSyncUtils {
+
+    private static final ShoppingListCustomActionBuilder shoppingListCustomActionBuilder =
+        ShoppingListCustomActionBuilder.of();
+
+    /**
+     * Compares all the fields of a {@link ShoppingList} and a {@link ShoppingListDraft}. It returns a {@link List} of
+     * {@link UpdateAction}&lt;{@link ShoppingList}&gt; as a result. If no update action is needed, for example in
+     * case where both the {@link ShoppingListDraft} and the {@link ShoppingList} have the same fields, an empty
+     * {@link List} is returned.
+     *
+     * @param oldShoppingList the shopping list which should be updated.
+     * @param newShoppingList the shopping list draft where we get the new data.
+     * @param syncOptions the sync options wrapper which contains options related to the sync process supplied
+     *                    by the user. For example, custom callbacks to call in case of warnings or errors occurring
+     *                    on the build update action process. And other options (See {@link ShoppingListSyncOptions}
+     *                    for more info.
+     * @return A list of shopping list specific update actions.
+     */
+    @Nonnull
+    public static List<UpdateAction<ShoppingList>> buildActions(
+        @Nonnull final ShoppingList oldShoppingList,
+        @Nonnull final ShoppingListDraft newShoppingList,
+        @Nonnull final ShoppingListSyncOptions syncOptions) {
+
+        final List<UpdateAction<ShoppingList>> updateActions = filterEmptyOptionals(
+            buildSetSlugUpdateAction(oldShoppingList, newShoppingList),
+            buildChangeNameUpdateAction(oldShoppingList, newShoppingList),
+            buildSetDescriptionUpdateAction(oldShoppingList, newShoppingList),
+            buildSetCustomerUpdateAction(oldShoppingList, newShoppingList),
+            buildSetAnonymousIdUpdateAction(oldShoppingList, newShoppingList),
+            buildSetDeleteDaysAfterLastModificationUpdateAction(oldShoppingList, newShoppingList)
+        );
+
+        final List<UpdateAction<ShoppingList>> shoppingListCustomUpdateActions =
+            buildPrimaryResourceCustomUpdateActions(oldShoppingList,
+                new CustomShoppingListDraft(newShoppingList),
+                shoppingListCustomActionBuilder,
+                syncOptions);
+
+        updateActions.addAll(shoppingListCustomUpdateActions);
+
+        return updateActions;
+    }
+
+    // TODO: https://github.com/commercetools/commercetools-jvm-sdk/issues/2073
```

Comment:
Can you also leave a comment on how it relates to this withKey implementation?

Reply:
Good point, I added the reason why I added this class: 
https://github.com/commercetools/commercetools-sync-java/pull/611/commits/a1468eae43b3088412645f2014b92b41bc674ca7